### ISSUE TITLE
retrieved hash chain is invalid: invalid difficulty: have

### ISSUE THREAD

**Issue by phpmac** [ethereum/go-ethereum#29810]
001===> 198000    set Difficulty 52 1 53 180087
WARN [05-21|08:11:48.755] Invalid header encountered               number=23      hash=8ac861..c6f780 parent=aae855..ddac02 err="invalid difficulty: have 180000, want 200000"
001===> 158776    set Difficulty 56 1 57 144272
001===> 174653    set Difficulty 46 1 47 158853
001===> 200000    set Difficulty 50 1 51 189919
WARN [05-21|08:11:48.755] Rewinding blockchain to block            target=0
001===> 200000    set Difficulty 40 1 41 200097
INFO [05-21|08:11:48.765] Loaded most recent local block           number=0       hash=e59f12..d6a19a td=131,072   age=55y2mo4d
INFO [05-21|08:11:48.765] Loaded last fast-sync pivot marker       number=396,974
WARN [05-21|08:11:48.765] Rolled back chain segment                header=68->0 snap=68->0 block=68->0 reason="invalid difficulty: have 180000, want 200000"
001===> 144179    set Difficulty 0 1 1 131072
INFO [05-21|08:11:48.766] Syncing: chain download in progress      synced=0.00% chain=18.00B headers=0@6.00B bodies=0@6.00B receipts=0@6.00B eta=-7.045s
WARN [05-21|08:11:48.766] Synchronisation failed, dropping peer    peer=c17f6c08b9be10802ab729949c73c474bb0854462c8ff68784740c91338f8b04 err="retrieved hash chain is invalid: invalid difficulty: have 180000, want 200000"
INFO [05-21|08:11:48.891] Commit new sealing work                  number=1       sealhash=4bfc0c..77a1eb uncles=0 txs=0 gas=0 fees=0 elapsed=125.059ms
001===> 144179    set Difficulty 0 1 1 131072
INFO [05-21|08:11:49.001] Commit new sealing work                  number=1       sealhash=1b126a..b9e0f3 uncles=0 txs=0 gas=0 fees=0 elapsed=110.633ms
INFO [05-21|08:11:49.099] Commit new sealing work                  number=1       sealhash=1b126a..b9e0f3 uncles=0 txs=0 gas=0 fees=0 elapsed=208.284ms
> INFO [05-21|08:11:58.662] Looking for peers                        peercount=0 tried=267 static=0

**Comment by s1na** [comment]
What kind of network is this? It's surely not mainnet or any of the public testnets.

**Comment by phpmac** [comment]
My own private chain network has modified the chainid.

2024 年 5 月 28 日星期二 19:21, Sina M ***@***.***(mailto:2024 年 5 月 28 日星期二 19:21, Sina M <<a href=)> 来信：

**Comment by s1na** [comment]
Ok it sounds like you have also modified the geth source?

The version must be outdated too. Geth doesn't support PoW mining anymore.

**Comment by phpmac** [comment]
Yes, I'm using the old version. It's a private chain. How to solve this problem? The block speed is also extremely fast. The disk is quickly occupied, and multiple nodes can't catch up synchronously. Do you know how to solve it?

2024 年 5 月 29 日星期三 03:05, Sina M ***@***.***(mailto:2024 年 5 月 29 日星期三 03:05, Sina M <<a href=)> 来信：

**Comment by MariusVanDerWijden** [comment]
Unfortunately we can not help you debug your fork of geth on your private chain
